drivers/lora/sx126x.c

```diff
@@ -0,0 +1,475 @@
+	CalibrationParams_t cal = {
+		.Value = SX126X_CALIBRATION_ALL,
+	};
+
+	LOG_DBG("TCXO on DIO3");
+
+	/* Delay in units of 15.625 us (1/64 ms) */
+	SX126xSetDio3AsTcxoCtrl(TCXO_DIO3_VOLTAGE,
+				TCXO_POWER_STARTUP_DELAY_MS << 6);
+	SX126xCalibrate(cal);
+#else
+	LOG_DBG("No TCXO configured");
+#endif
+}
+
+void SX126xReset(void)
+{
+	LOG_DBG("Resetting radio");
+	gpio_pin_set(dev_data.reset, GPIO_RESET_PIN, 1);
+	k_sleep(K_MSEC(20));
+	gpio_pin_set(dev_data.reset, GPIO_RESET_PIN, 0);
+	k_sleep(K_MSEC(10));
+}
+
+void SX126xSetRfTxPower(int8_t power)
+{
+	LOG_DBG("power: %" PRIi8, power);
+	SX126xSetTxParams(power, RADIO_RAMP_40_US);
+}
+
+void SX126xWaitOnBusy(void)
+{
+	while (gpio_pin_get(dev_data.busy, GPIO_BUSY_PIN)) {
+		k_sleep(K_MSEC(1));
+	}
+}
+
+void SX126xWakeup(void)
+{
+	int ret;
+
+	uint8_t req[] = { RADIO_GET_STATUS, 0 };
+	const struct spi_buf tx_buf = {
+		.buf = req,
+		.len = sizeof(req),
+	};
+
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1,
+	};
+
+	LOG_DBG("Sending GET_STATUS");
+	ret = spi_write(dev_data.spi, &dev_data.spi_cfg, &tx);
+	if (ret < 0) {
+		LOG_ERR("SPI transaction failed: %i", ret);
+		return;
+	}
+
+	LOG_DBG("Waiting for device...");
+	SX126xWaitOnBusy();
+	LOG_DBG("Device ready");
+}
+
+static void sx126x_dio1_irq_work_handler(struct k_work *work)
+{
+	LOG_DBG("Processing IRQ");
+	if (!dev_data.radio_dio_irq) {
+		LOG_WRN("Radio IRQ before setting radio_dio_irq.");
+		return;
+	}
+
+	dev_data.radio_dio_irq(NULL);
+	if (Radio.IrqProcess) {
+		Radio.IrqProcess();
+	}
+}
+
+static void sx126x_dio1_irq_callback(struct device *dev,
+				     struct gpio_callback *cb, uint32_t pins)
+{
+	if (pins & BIT(GPIO_DIO1_PIN)) {
+		k_work_submit(&dev_data.dio1_irq_work);
+	}
+}
+
+static int sx126x_lora_init(struct device *dev)
+{
+	int ret;
+
+	LOG_DBG("Initializing %s", DT_INST_LABEL(0));
+
+	dev_data.reset = device_get_binding(GPIO_RESET_LABEL);
+	if (!dev_data.reset) {
+		LOG_ERR("Cannot get pointer to %s device", GPIO_RESET_LABEL);
+		return -EIO;
+	}
+
+	gpio_pin_configure(dev_data.reset, GPIO_RESET_PIN,
+			   GPIO_OUTPUT_ACTIVE | GPIO_RESET_FLAGS);
+
+
+	dev_data.busy = device_get_binding(GPIO_BUSY_LABEL);
+	if (!dev_data.busy) {
+		LOG_ERR("Cannot get pointer to %s device", GPIO_BUSY_LABEL);
+		return -EIO;
+	}
+
+	gpio_pin_configure(dev_data.busy, GPIO_BUSY_PIN,
+			   GPIO_INPUT | GPIO_BUSY_FLAGS);
+
+	dev_data.dio1 = device_get_binding(GPIO_DIO1_LABEL);
+	if (!dev_data.dio1) {
+		LOG_ERR("Cannot get pointer to %s device", GPIO_DIO1_LABEL);
+		return -EIO;
+	}
+
+	gpio_pin_configure(dev_data.dio1, GPIO_DIO1_PIN,
+			   GPIO_INPUT | GPIO_INT_DEBOUNCE | GPIO_DIO1_FLAGS);
+
+	k_work_init(&dev_data.dio1_irq_work, sx126x_dio1_irq_work_handler);
+
+	gpio_init_callback(&dev_data.dio1_irq_callback,
+			   sx126x_dio1_irq_callback, BIT(GPIO_DIO1_PIN));
+	if (gpio_add_callback(dev_data.dio1, &dev_data.dio1_irq_callback) < 0) {
+		LOG_ERR("Could not set GPIO callback for IRQ line.");
```

Comment:
Showing the IRQ line number would be helpful here.

Reply:
I have rephrased this as "Could not set GPIO callback for DIO1 interrupt" I think that should be a bit clearer.